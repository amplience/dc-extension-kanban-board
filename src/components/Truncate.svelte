<script lang="ts">
  export let text: string = '';
</script>

<style lang="scss">
  .truncate {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }
</style>

<div class="truncate" title={text}>{text}</div>
