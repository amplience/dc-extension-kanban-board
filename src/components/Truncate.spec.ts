import { render } from '@testing-library/svelte';
import Truncate from './Truncate.svelte';

describe('truncate component', () => {
  it('should render a truncate component with value', () => {
      const { container } = render(Truncate, {
        text: 'TEXT' 
      });

      expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a truncate component with default value', () => {
      const { container } = render(Truncate, {
        // no-op.
      });

      expect(container.firstChild).toMatchSnapshot();
  });
});
