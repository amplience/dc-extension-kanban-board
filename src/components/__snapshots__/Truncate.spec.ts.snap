// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`truncate component should render a truncate component with default value 1`] = `
<div>
  <div
    class="truncate svelte-atm87y"
    title=""
  >
    
  </div>
</div>
`;

exports[`truncate component should render a truncate component with value 1`] = `
<div>
  <div
    class="truncate svelte-atm87y"
    title="TEXT"
  >
    TEXT
  </div>
</div>
`;
